.github/workflows/tests_gpu.yml

```diff
@@ -0,0 +1,300 @@
+                          -DKokkos_ENABLE_${{ matrix.exec_model }}=ON \
+                          -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF \
+                          -DCMAKE_CXX_COMPILER="$(which g++-$GCC_VERSION)" \
+                          -DCMAKE_CXX_STANDARD=20 \
+                          -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
+                          -G Ninja
+          cmake --build ./Build --verbose
+          cmake --install ./Build
+          cd -
+
+      - name: Test Kokkos core library (Ubuntu)
+        env: 
+          CMAKE_PREFIX_PATH: ${{ github.workspace }}/Kokkos_install/${{ matrix.exec_model }}
+        run: |
+          set -x
+          sudo apt-get update && sudo apt-get -y -q install gfortran-$GCC_VERSION
+          cd kokkos_${{ matrix.exec_model }}/example/build_cmake_installed
+          which -a gfortran || which -a gfortran-$GCC_VERSION
+          cmake -B build -DCMAKE_Fortran_COMPILER=gfortran-$GCC_VERSION
+          cmake --build build
+          cmake --build build --target test
+          set +x
+
+      - name: Cache installation directories
+        id: kokkos-cache
+        uses: actions/cache/save@v3
+        with:
+          key: ${{ matrix.os }}-kokkos${{ matrix.kokkos_version }}-${{ matrix.exec_model }}-${{ github.ref }}-${{ github.sha }}
+          path: ${{ github.workspace }}/Kokkos_install/${{ matrix.exec_model }}
+
+  cpptestswithKokkos:
+    if: ${{ github.sha == 'xxxxxxxx' }}
+    needs: [builddeps]
+    strategy:
+      matrix:
+        os: [ubuntu-22.04]
+        pl_backend: ["lightning_kokkos"]
+        exec_model: ["CUDA"]
+        kokkos_version: ["4.1.00"]
+
+    name: C++ tests (Kokkos)
+    runs-on: 
+      - ${{ matrix.os }}
+      - self-hosted
+      - gpu
+      
+    steps:
+      - name: Validate GPU version and installed compiler
+        run: |
+          source /etc/profile.d/modules.sh
+          module use /opt/modules
+          module load cuda/12.2
+          echo "${PATH}" >> $GITHUB_PATH
+          echo "LD_LIBRARY_PATH=${LD_LIBRARY_PATH}" >> $GITHUB_ENV
+          nvcc --version
+          nvidia-smi
+
+      - name: Restoring cached dependencies
+        id: kokkos-cache
+        uses: actions/cache/restore@v3
+        with:
+          path: ${{ github.workspace }}/Kokkos_install/${{ matrix.exec_model }}
+          key: ${{ matrix.os }}-kokkos${{ matrix.kokkos_version }}-${{ matrix.exec_model }}-${{ github.ref }}-${{ github.sha }}
+          restore-keys: ${{ matrix.os }}-kokkos${{ matrix.kokkos_version }}-${{ matrix.exec_model }}-${{ github.ref }}-
+
+      - name: Copy cached libraries
+        run: |
+          mkdir -p ${{ github.workspace }}/Kokkos
+          ls -ltrh
+          cp -r ${{ github.workspace }}/Kokkos_install/${{ matrix.exec_model }}/* ${{ github.workspace }}/Kokkos
+
+      - name: Checkout PennyLane-Lightning
+        uses: actions/checkout@v3
+        with:
+          path: main
+          fetch-depth: 2
```

Comment:
Should we go for the default value here?

Reply:
Yes.